charts/redpanda/templates/secrets.yaml

```diff
@@ -62,33 +65,33 @@ stringData:
           status=$(${CURL_MAINTENANCE_DELETE_CMD})
           sleep 0.5
       done
-    }
 
-    export -f postStartHook
-    timeout {{ div .Values.statefulset.terminationGracePeriodSeconds 2 }} bash -c "postStartHook"
-    true
+{{- if and .Values.auth.sasl.enabled (not (empty .Values.auth.sasl.secretRef )) }}
+      # Setup and export SASL bootstrap-user
+      IFS=":" read -r USER_NAME PASSWORD MECHANISM < $(find /etc/secrets/users/* -print)
+      MECHANISM=${MECHANISM:-{{- include "sasl-mechanism" . }}}
+      rpk acl user create ${USER_NAME} --password=${PASSWORD} --mechanism ${MECHANISM} {{ template "rpk-flags-no-brokers-no-sasl" $ }} || true
 {{- end }}
 
-{{- if and .Values.auth.sasl.enabled (not (empty .Values.auth.sasl.secretRef )) }}
-    set +x
+      touch /tmp/postStartHookFinished
+    }
+
+    postStartHook
```

Comment:
Do we need to include the check ```{{- if gt ( .Values.statefulset.replicas | int64 ) 2 }}``` here also, as you do in the post start?

Reply:
No. There's no harm in trying to take it _out_ of maintenance mode. Maybe it got there when scaling down or something.